lib/spack/spack/util/windows_registry.py

```diff
@@ -68,8 +69,20 @@ def _gather_subkey_info(self):
         sub_keys, _, _ = winreg.QueryInfoKey(self.hkey)
         for i in range(sub_keys):
             sub_name = winreg.EnumKey(self.hkey, i)
-            sub_handle = winreg.OpenKeyEx(self.hkey, sub_name, access=winreg.KEY_READ)
-            self._keys.append(RegistryKey(os.path.join(self.path, sub_name), sub_handle))
+            try:
+                sub_handle = winreg.OpenKeyEx(self.hkey, sub_name, access=winreg.KEY_READ)
+                self._keys.append(RegistryKey(os.path.join(self.path, sub_name), sub_handle))
+            except OSError as e:
+                if hasattr(e, "winerror"):
+                    if e.winerror == 5:
+                        # This is a permission error, we can't read this key
+                        # move on
+                        # TODO: johnwparent add a flag toggling whether we throw here or not
```

Comment:
Some registry keys are inaccessible without elevated permissions consistently due to Windows restrictions whereas some may be arbitrarily inaccessible due to admin/organization policy governing user access to the registry, so the answer to "Is it unexpected that some keys may be unreadable", is that, it may be given any number of circumstances. The method is not used to search for  specific subkeys, but is instead used to return all available subkeys associated with a given registry key, so we have no real way to determine in this scope what the caller was actually searching for as we just return a list of all keys. From my perspective, storing a list of all the inaccessible keys (which we would have to represent as string paths to the key as we cannot create a handle to the key) would require the caller to make yet another call into the `RegistryKey` object that just returned the initial list or complicate the return type, which IMO is more of a burden/invasive. However, this method is also at the moment, a private method only used by the `keys` property, which, conceptually I would expect to return ALL keys, and if there was a key absent, I would assume it didn't exist rather than it being inaccessible (which can mean something else), but I also don't think that property should return a tuple with `keys` and `inaccessible` keys. Maybe another method, something like `readable_keys` that doesn't throw?

Reply:
I think this all makes sense, but my follow-up thought is that given all this it doesn't make sense to ever throw the exception here precisely (and hence the TODO comment doesn't make sense to me).

Reply:
Agreed, the TODO was more of a suggestion that I wanted to be sure to address during review anyway.